xclim/core/units.py

```diff
@@ -285,60 +287,79 @@ def convert_units_to(
 
     Returns
     -------
-    xr.DataArray or float or int or str or Any
+    str or xr.DataArray or units.Unit or units.Quantity
         The source value converted to target's units.
+        The outputted type is always similar to `source` initial type.
     """
     # Target units
     if isinstance(target, units.Unit):
-        tu = target
+        target_unit = target
     elif isinstance(target, (str, xr.DataArray)):
-        tu = units2pint(target)
+        target_unit = units2pint(target)
     else:
-        raise NotImplementedError
+        raise NotImplementedError(
+            "target must be either a pint Unit or a xarray DataArray."
+        )
 
     if isinstance(source, str):
         q = str2pint(source)
         # Return magnitude of converted quantity. This is going to fail if units are not compatible.
-        return q.to(tu).m
+        return q.to(target_unit).m
 
     if isinstance(source, units.Quantity):
-        return source.to(tu).m
+        return source.to(target_unit).m
 
     if isinstance(source, xr.DataArray):
-        fu = units2pint(source)
-        tu_u = pint2cfunits(tu)
+        source_unit = units2pint(source)
+        target_cf_unit = pint2cfunits(target_unit)
 
-        if fu == tu:
+        if source_unit == target_unit:
             # The units are the same, but the symbol may not be.
-            source.attrs["units"] = tu_u
+            source.attrs["units"] = target_cf_unit
             return source
 
         with units.context(context or "none"):
+            if (
+                _is_precipitation_amount(source, source_unit)
+                and target_unit.dimensionality.get("[time]") == -1
```

Comment:
Would it be too strict to do:
`_is_precipitation(source) and _is_amount(source_unit) and _is_rate(target_unit)`
?

This would indeed allow, to have a reverse operation, if ever needed.

Reply:
I suggest we keep the reverse operation for another PR if needed.